src/coord_seq.rs

```diff
@@ -53,6 +61,80 @@ impl<'a> CoordSeq<'a> {
         }
     }
 
+    /// Creates a new `CoordSeq`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::CoordSeq;
+    ///
+    /// let coords = CoordSeq::new_from_vec(&[&[0., 1.], &[2., 3.]])
+    ///                       .expect("failed to create CoordSeq");
+    /// assert!(coords.get_y(1) == Ok(3.));
+    ///
+    /// // Doing it from a Vec<Vec<f64>>.
+    /// let positions = vec![vec![0., 1.], vec![2., 3.]];
+    /// let s_positions = positions.iter().map(|x| x.as_slice()).collect::<Vec<_>>();
+    /// let coords = CoordSeq::new_from_vec(&s_positions)
+    ///                       .expect("failed to create CoordSeq");
+    /// assert!(coords.get_y(1) == Ok(3.));
+    ///
+    /// // All vectors don't have the same length, this is an error!
+    /// assert!(CoordSeq::new_from_vec(&[&[0., 1.], &[3.]]).is_err());
+    ///
+    /// // An empty vector is an error as well since we can't figure out its dimensions!
+    /// assert!(CoordSeq::new_from_vec(&[]).is_err());
+    /// ```
+    pub fn new_from_vec(data: &[&[f64]]) -> GResult<CoordSeq<'a>> {
+        let size = data.len();
+
+        if size > 0 {
+            let dims = data[0].len();
+            if let Err(e) = CoordDimensions::try_from(dims as _) {
+                return Err(Error::GenericError(e.to_owned()));
+            }
+            if !data.iter().skip(1).all(|x| x.len() == dims) {
+                return Err(Error::GenericError("All vec entries must have the same size!".to_owned()));
+            }
+            match match GContextHandle::init() {
+                Ok(context_handle) => {
+                    unsafe {
+                        let ptr = GEOSCoordSeq_create_r(context_handle.as_raw(),
+                                                        size as _,
+                                                        dims as _);
+                        CoordSeq::new_from_raw(ptr, Arc::new(context_handle), size as _, dims as _)
+                    }
+                }
+                Err(e) => return Err(e),
+            } {
+                Ok(coord) => {
+                    let raw_context = coord.get_raw_context();
+                    let raw_coord = coord.as_raw();
+
+                    let funcs = [GEOSCoordSeq_setX_r,
+                                 GEOSCoordSeq_setY_r,
+                                 GEOSCoordSeq_setZ_r];
+
+                    for (line, line_data) in data.iter().enumerate() {
+                        for (pos, elem) in line_data.iter().enumerate() {
```

Comment:
note sure about this, but shouldn't you check than there are never more than 3 elements ?

Reply:
It's done before when I check the dimensions of each line.